### DISCUSSION TITLE
installing antismash via docker - mem/swap requirements not documented?

### DISCUSSION THREAD

**Discussion by AjitPS** [antismash/antismash#510] [General]
Would be useful to know memory requirements to pull an antismash docker image to a linux VM and run it. The dockerfile https://github.com/antismash/antismash/blob/master/Dockerfile and docker run https://github.com/antismash/antismash/blob/master/docker/run_from_docker script seem pretty straightforward but docs https://docs.antismash.secondarymetabolites.org/install/#antismash-standalone make no mention of memory, swap, port etc.

**Comment by SJShaw** [comment]
Ports aren't mentioned because they aren't relevant at all for the antiSMASH docker container, there is no network communication of any kind.

If you need to use swap, your performance may suffer enormously. Whether you want to allow it to be used is entirely up to you and the system you'll be running the container on. There's nothing in antiSMASH (or any software, I'd hope) that requires swap to exist.

As for memory, well, there's no way to give a meaningful number for memory use that covers any variety of use cases. Input files can range from single clusters (in the order of kilobytes) to metagenomic dumps (in the order of gigabytes). The amount of memory required to process those will also depend on the complexity and assembly quality, to a degree, of the particular input. A small input with very dense clusters throughout the input may probably need more memory than a very sparsely populated larger input.

I've yet to have an issue processing full bacterial genomes on a laptop with 16GB of RAM, but your best bet is to benchmark with a few inputs you'd consider typical for your use case (both in terms of inputs themselves and analysis options) and see how much memory you end up using.
